http-server/src/test/java/org/triplea/server/http/DropwizardTest.java

```diff
@@ -0,0 +1,51 @@
+package org.triplea.server.http;
+
+import static org.junit.jupiter.api.extension.ExtensionContext.Namespace.GLOBAL;
+
+import com.github.database.rider.core.api.dataset.DataSet;
+import com.github.database.rider.junit5.DBUnitExtension;
+import io.dropwizard.testing.DropwizardTestSupport;
+import java.net.URI;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.triplea.test.common.Integration;
+
+/** Core configuration for a test that will start a dropwizard server and initialize database. */
+@Integration
+@DataSet(cleanBefore = true, value = "integration.yml")
+@ExtendWith(DBUnitExtension.class)
+@ExtendWith(DropwizardTest.DropwizardServerExtension.class)
+abstract class DropwizardTest {
+  final URI localhost = URI.create("http://localhost:8080");
+
+  /**
+   * Extension to start a drop wizard server before all tests and then shuts it down afterwards.
+   * Note, if a server is already running, then that server is used.
+   */
+  public static class DropwizardServerExtension
+      implements BeforeAllCallback, ExtensionContext.Store.CloseableResource {
+    private static boolean started = false;
+
+    private static final DropwizardTestSupport<AppConfig> support =
+        new DropwizardTestSupport<>(ServerApplication.class, "configuration-prerelease.yml");
+
+    @Override
+    public void beforeAll(final ExtensionContext context) {
+      if (!started) {
+        started = true;
+        try {
+          support.before();
+        } catch (final RuntimeException e) {
+          // ignore, server is already started
+        }
+        context.getRoot().getStore(GLOBAL).put("dropwizard-startup", this);
+      }
+    }
+
+    @Override
+    public void close() {
+      support.after();
```

Comment:
TBH I have no Idea how the JUnit5 extension model works, but how do you tell that close is only called after all tests were executed and why isn't there a similar thing for instantiation?
The [documentation](https://junit.org/junit5/docs/current/user-guide/#extensions-keeping-state) suggests that the constructor might be the corresponding initializer

Reply:
Seems to be the magic of the extension. When this code was originally introduced it was tested and we've been using it for some time now too; so there it's battle-tested and we can have some confidence it works. More practically, removing the context registration call and this is no longer called, so it's Junit5 extension magic happening here.